### PR TITLE
Remove p8-platform

### PR DESCRIPTION
**Pull request by phunkyfish**
v8.0.1
- Remove p8-platform dependency
- Remove charset converter dependency
- Use std::thread, std::mutex, condition_variable instead of event and bool defines
- Use thread sleep_for instead of p8 time utils
- Remove SAFE_DELETE
- Use kodi StringUtils
- Remove p8 os includes
- Revert "Removed no more supported (C++17) declaration keyword (register)"

### CODE REVIEW

**Comment by phunkyfish** [comment]
@AlwinEsch also ready for a runtime test.

File: src/lib/tsreader/MemoryBuffer.cpp
Comment (AlwinEsch) [inline src/lib/tsreader/MemoryBuffer.cpp]: About here, thing the event wait is about milliseconds, with 5000 seconds a bit long,

File: src/lib/tsreader/MemoryBuffer.cpp
Comment (AlwinEsch) [inline src/lib/tsreader/MemoryBuffer.cpp]: About here and others makes `SafeDelete` not direct sense as on them before `if (item->data)` the `item` can be empty.

Was then also bad before with `SAFE_DELETE_ARRAY`. Also a NULL set not needed as the array cleared after it.

Think most correct would be (maybe also with new `for`):
```cpp
  for (auto& item : m_Array)
  {
    if (item)
    {
      if (item->data)
        delete[] item->data;
      delete item;
    }
  }

  m_Array.clear();
```

File: src/lib/tsreader/MemoryBuffer.cpp
Comment (AlwinEsch) [inline src/lib/tsreader/MemoryBuffer.cpp]: For here e.g. like comment before.

File: src/lib/tsreader/TSReader.cpp
Comment (AlwinEsch) [inline src/lib/tsreader/TSReader.cpp:240]: You changed on the others to bool, only is in this class still `long` and e.g. `E_FAIL` used (where 1, the `S_OK` is 0).

Also on e.g. `long retval = m_fileReader->OpenFile(m_fileName);` becomes the boolean return checked with old.

File: src/os-dependent.h
Comment (AlwinEsch) [inline src/os-dependent.h:132]: Maybe we can prevent this header too.

**Review by AlwinEsch — CHANGES_REQUESTED** [review]
There is a lot where not seems to match, Sorry!

Will also look tommorow about here.

File: src/lib/tsreader/MemoryBuffer.cpp
Comment (phunkyfish) [reply src/lib/tsreader/MemoryBuffer.cpp]: Whoops!

File: src/os-dependent.h
Comment (margro) [reply src/os-dependent.h:132]: Please don't! The TSReader code is partially coming from Mediaportal. I cannot merge improvements if you're going to change the interface of these functions. Keep it.

**Comment by margro** [comment]
I'm not happy with the changes to the tsreader code. It makes importing fixes from MediaPortal more difficult.
There is a good reason why I'm keeping the Windows return values and macros in here...

File: src/os-dependent.h
Comment (phunkyfish) [reply src/os-dependent.h:132]: Ok, understood. So no modifications to anything in lib?

File: src/os-dependent.h
Comment (margro) [inline src/os-dependent.h]: Please keep this! Is is essential since a Windows wchar_t != Linux wchar_t
The file I'm reading is in Windows format (2-bytes) and a Linux wchar_t is 4 bytes.

File: src/windows/FileUtils.cpp
Comment (margro) [inline src/windows/FileUtils.cpp]: Wrong.  SHGetKnownFolderPath is a Windows function and it does not return a bool.

File: src/os-dependent.h
Comment (margro) [reply src/os-dependent.h:132]: Preferably not. I'm ok with the removal of the p8-platform code, but keep the os-dependent.h
And before merging your changes to will need run-time testing to make sure that we're not breaking the video playback.

Comment (phunkyfish) [reply src/os-dependent.h:132]: Ok, sounds good. I will post back on this PR once I have restored lib folder back excluding the p8 changes of course.

File: src/os-dependent.h
Comment (phunkyfish) [reply src/os-dependent.h]: Will do

File: src/windows/FileUtils.cpp
Comment (phunkyfish) [reply src/windows/FileUtils.cpp]: Will revert this.

**Comment by phunkyfish** [comment]
@margro if you could review again please. It should be cleaned up with minimal change to the lib folder now.

FYI, the jenkins error is not related to this PR.

File: pvr.mediaportal.tvserver/changelog.txt
Comment (AlwinEsch) [inline pvr.mediaportal.tvserver/changelog.txt]: Here you have a rebase fault.

File: src/lib/tsreader/DvbUtil.cpp
Comment (AlwinEsch) [inline src/lib/tsreader/DvbUtil.cpp]: Was a reason to add regster? Seen since C++11 deprecated and with C++17 no more used.

**Review by AlwinEsch — COMMENTED** [review]
From overview it looks good, will test later today to confirm correct work (if I become my Windows happy again :sweat_smile:).

File: pvr.mediaportal.tvserver/changelog.txt
Comment (phunkyfish) [reply pvr.mediaportal.tvserver/changelog.txt]: Whoops

File: src/lib/tsreader/DvbUtil.cpp
Comment (phunkyfish) [reply src/lib/tsreader/DvbUtil.cpp]: Not sure at all here.

Comment (phunkyfish) [reply src/lib/tsreader/DvbUtil.cpp]: Ok, removed.

**Comment by phunkyfish** [comment]
@margro it appears the problem that blocked us in testing this was an issue in mediaportal itself. We assume all users of the addon already had a workaround for it. The addon needed port 9596, but on Windows Firewall only 9591 was open.

We will merge this in a couple of days as it is blocking branching addons for the next kodi version and also the switch to cpp17.

All of your concerns should have been addressed in the previous push.

File: src/lib/tsreader/MemoryBuffer.cpp
Comment (AlwinEsch) [inline src/lib/tsreader/MemoryBuffer.cpp]: This line with "it" can be removed, too.

**Review by AlwinEsch — APPROVED** [review]
Except the one part, is this request OK.

Has done runtime tests and in compare between here and version before and it works the same. Hope to not overseen something :sweat_smile: 

There are some bugs by addon and also by server, but there must come a good maintainer where know both places and look about fixes.

File: src/lib/tsreader/MemoryBuffer.cpp
Comment (phunkyfish) [reply src/lib/tsreader/MemoryBuffer.cpp]: We can only make minimal changes to anything in the lib folder.

**Pull request merged**
